### ISSUE TITLE
RT-145462: cpan(1) log message uses the wrong key 

### ISSUE THREAD

**Issue by briandfoy** [andk/cpanpm#167]
I'm transferring and splitting up some issues reported in [App::Cpan many problems, multiple fixes](https://rt.cpan.org/Ticket/Display.html?id=145462).

In App::Cpan::_download, the log line uses the wrong variable as a hash key:

    $logger->info( "Downloaded [$arg] to [$paths{$arg}]" );

It should be:

    $logger->info( "Downloaded [$arg] to [$paths{$module}]" );

This is related to #162 with some rearranging that needs to happen for track errors and return the appropriate values.
 
Patch provided by "eponymous alias" <eponymousalias@yahoo.com> in the RT ticket
[App.Cpan.patch.txt](https://github.com/andk/cpanpm/files/10180672/App.Cpan.patch.txt)

**Comment by x-yuri** [comment]
Indeed, right now `cpan -g` exits with an error (although the operation succeeds). The issue seemed to be [fixed][a], but soon [broken again][b].

[a]: https://github.com/andk/cpanpm/commit/3fde9b869971f90910145afe947071c5e8acda96
[b]: https://github.com/andk/cpanpm/commit/49ac94eab01d2d41e08dde187855c7acecd1c1f9
